interlok-core/src/main/java/com/adaptris/security/password/SeededAesPbeCrypto.java

```diff
@@ -0,0 +1,164 @@
+  {
+    String encryptedString = encrypted;
+    if (encrypted.startsWith(SEEDED_BATCH))
+    {
+      encryptedString = encrypted.substring(SEEDED_BATCH.length());
+    }
+    try
+    {
+      byte[] secret = seedAndIv();
+      byte[] iv = ArrayUtils.subarray(secret, 0, IV_LENGTH);
+      char[] seed = new HexStringByteTranslator().translate(ArrayUtils.subarray(secret, IV_LENGTH, secret.length)).toCharArray();
+
+      PBEParameterSpec pbeParamSpec = new PBEParameterSpec(SALT, ITERATIONS, new IvParameterSpec(iv));
+      PBEKeySpec pbeKeySpec = new PBEKeySpec(seed);
+      SecretKeyFactory keyFac = SecretKeyFactory.getInstance(ALGORITHM);
+      SecretKey pbeKey = keyFac.generateSecret(pbeKeySpec);
+      Cipher pbeCipher = Cipher.getInstance(ALGORITHM);
+      pbeCipher.init(Cipher.DECRYPT_MODE, pbeKey, pbeParamSpec);
+
+      byte[] decrypted = pbeCipher.doFinal(base64.translate(encryptedString));
+      return unseed(decrypted, charset);
+    }
+    catch (Exception e)
+    {
+      throw new PasswordException(e);
+    }
+  }
+
+  @Override
+  public String encode(String plainText, String charset) throws PasswordException
+  {
+    try
+    {
+      byte[] secret = seedAndIv();
+      byte[] iv = ArrayUtils.subarray(secret, 0, IV_LENGTH);
+      char[] seed = new HexStringByteTranslator().translate(ArrayUtils.subarray(secret, IV_LENGTH, secret.length)).toCharArray();
+
+      PBEParameterSpec pbeParamSpec = new PBEParameterSpec(SALT, ITERATIONS, new IvParameterSpec(iv));
+      PBEKeySpec pbeKeySpec = new PBEKeySpec(seed);
+      SecretKeyFactory keyFac = SecretKeyFactory.getInstance(ALGORITHM);
+      SecretKey pbeKey = keyFac.generateSecret(pbeKeySpec);
+      Cipher pbeCipher = Cipher.getInstance(ALGORITHM);
+      pbeCipher.init(Cipher.ENCRYPT_MODE, pbeKey, pbeParamSpec);
+
+      byte[] encrypted = pbeCipher.doFinal(seed(plainText, charset));
+      return SEEDED_BATCH  + base64.translate(encrypted);
+    }
+    catch (Exception e)
+    {
+      throw new PasswordException(e);
+    }
+  }
+
+  private byte[] seedAndIv() throws PasswordException
+  {
+    SHA384Digest sha = new SHA384Digest();
+    try
+    {
+      if (seedFile == null)
+      {
+        if ((seedFile = System.getProperty("password.seed")) == null)
+        {
+          StringBuffer sb = new StringBuffer();
+          for (String k : System.getProperties().stringPropertyNames())
+          {
+            sb.append(k).append(" = ").append(System.getProperty(k)).append('\n');
+          }
+          System.err.println("Properties = " + sb);
+
+          throw new PasswordException("No seed file specified in system properties \"password.seed\"");
+        }
+      }
+      try (FileInputStream in = new FileInputStream(seedFile))
+      {
+        byte[] b = new byte[1048576];
+        int z = 0;
+        while ((z = in.read(b)) > 0)
+        {
+          sha.update(b, 0, z);
+        }
+      }
```

Comment:
while relying on apache is a trifle foolish in the wake of log4shell ;) commons.io means we don't necessarily have to write our own loops.
- Is someone's seed file going to be massive, of course, if someone uses a 10GB seed file, this is a silly idea.

```
    try (FileInputStream in = new FileInputStream(f)) {
      sha.update(IOUtils.readFully(in, in.available()));
    }
```